### PR TITLE
Macos fixes

### PR DESCRIPTION
**Pull request by SadiinsoSnowfall**
while running the test on my apple M3-based machine, I found some errors regarding ULP and clang regressions.

### CODE REVIEW

File: test/unit/module/algo/algorithm/sums_special_cases.cpp [hunk 1]
```diff
@@ -143,9 +143,9 @@ TTS_CASE("eve.algo.transform_reduce, stateful map/reduce")
             eve::algo::transform_reduce(
                 v,
                 [coeff](auto x) { return x * coeff; },
-                std::pair {[_ = coeff](auto x, auto y)
+                std::pair {[state = coeff](auto x, auto y)
                            {
-                             (void)_;
+                             (void)state;
```
Comment (DenisYaroshevskiy) [inline test/unit/module/algo/algorithm/sums_special_cases.cpp:148]: This is reasonable, I don't know about tolerances increase.

Comment (jfalcou) [reply test/unit/module/algo/algorithm/sums_special_cases.cpp:148]: 1 ULPs sounds fait on those large functions. I'll cross check with jtlap

Comment (jfalcou) [reply test/unit/module/algo/algorithm/sums_special_cases.cpp:148]: @DenisYaroshevskiy @jtlap says it is fine, so I am gonna approve if it is OK for you.

**Comment by jtlap** [comment]
1ulp is not a problem here

**Pull request merged**